### PR TITLE
Variables: add variable support

### PR DESCRIPTION
**Pull request by ryantxu**
support for query variables. Related to #113 

image

### CODE REVIEW

**Comment by ryantxu** [comment]
@bymattoa -- most of this PR is formatting changes due to an upgrade in prettier :(

File: src/variables.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  query = (request: DataQueryRequest<SitewiseQuery>) => {
+    const { targets } = request;
+    if (targets && targets.length === 1) {
+      const query = targets[0];
+      if (query.queryType === QueryType.ListAssets || query.queryType === QueryType.ListAssetModels) {
+        return this.datasource.query(request).pipe(
+          mergeMap((rsp) => {
+            const assets = new DataFrameView<HierarchyInfo>(rsp.data[0]);
+            const data: MetricFindValue[] = assets.map((a) => ({
```
Comment (ryantxu) [inline src/variables.ts:25]: The default behavior just uses the string values as the variables... BUT for this, I assume we actually want the ids as the template value and the name as what is displayed

Comment (bymattoa) [reply src/variables.ts:25]: lgtm

**Review by bymattoa — COMMENTED** [review]
We're also going to need to fetch property IDs (which are available on models) as well as asset IDs. How would a user go about mapping these?

File: src/variables.ts (hunk 1 above)
Comment (bymattoa) [reply src/variables.ts:25]: We will also need to fetch property IDs. (See thread in conversation)

**Comment by ryantxu** [comment]
This now uses the "Describe Asset" query to pick properties:

image

**Comment by toddtreece** [comment]
i think #134 and #127 cover the changes needed for #113. closing this one

**Pull request closed without merging**